extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleSetTableOwner.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.plugin.spark.authz.ranger
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.analysis.{PersistedView, ViewType}
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.{Command, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.command.RunnableCommand
+import org.apache.spark.sql.execution.datasources.CreateTable
+
+import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
+import org.apache.kyuubi.plugin.spark.authz.util.WithInternalChildren
+
+class RuleSetTableOwner(spark: SparkSession) extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    val authzUser = getAuthzUgi(spark.sparkContext).getShortUserName
+    val v1SessionCatalog = spark.sessionState.catalog
+
+    plan match {
+      case CreateTable(tableDesc, _, _) if tableDesc.owner != authzUser =>
+        setFieldVal(plan, "tableDesc", tableDesc.copy(owner = authzUser))
+        plan
+      case r: RunnableCommand if r.nodeName == "CreateTableLikeCommand" =>
+        val tableId = getFieldVal[TableIdentifier](r, "targetTable")
+        SetTableOwnerCommand(r, tableId, authzUser, v1SessionCatalog.tableExists(tableId))
+      case r: RunnableCommand
+          if r.nodeName == "CreateViewCommand" && getFieldVal[ViewType](
+            r,
+            "viewType") == PersistedView =>
+        val tableId = getFieldVal[TableIdentifier](r, "name")
+        val replace = getFieldVal[Boolean](r, "replace")
+        SetTableOwnerCommand(r, tableId, authzUser, v1SessionCatalog.tableExists(tableId), replace)
+      case _ =>
+        plan
+    }
+  }
+}
+
+case class SetTableOwnerCommand(
```

Comment:
Shall we use `Authz SetTableOwnerCommand` for class name to  distinguish commands of authz's and spark's ? This command will be essential to control wrapping plans.

Reply:
Done